File: sdks/go/pkg/models/operations/postusersidemailaliases.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PostUsersIDEmailAliasesPathParams struct {
	UserID string `pathParam:"style=simple,explode=false,name=user_id"`
}

type PostUsersIDEmailAliasesRequestBody struct {
	Email string `json:"email"`
}

type PostUsersIDEmailAliasesRequest struct {
	Retries    *utils.RetryConfig
	PathParams PostUsersIDEmailAliasesPathParams
	Request    *PostUsersIDEmailAliasesRequestBody `request:"mediaType=application/json"`
}

type PostUsersIDEmailAliasesResponse struct {
	ClientError *shared.ClientError
	ContentType string
	EmailAlias  *shared.EmailAlias
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/postusersidemailaliases.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PostUsersIdEmailAliasesPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=user_id" })
  userId: string;
}


export class PostUsersIdEmailAliasesRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=email" })
  email: string;
}


export class PostUsersIdEmailAliasesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PostUsersIdEmailAliasesPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostUsersIdEmailAliasesRequestBody;
}


export class PostUsersIdEmailAliasesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  emailAlias?: shared.EmailAlias;

  @SpeakeasyMetadata()
  statusCode: number;
}
